libretroshare/src/pqi/authssl.cc

```diff
@@ -1198,6 +1198,21 @@ int AuthSSLimpl::VerifyX509Callback(int /*preverify_ok*/, X509_STORE_CTX* ctx)
 	std::string sslCn = RsX509Cert::getCertIssuerString(*x509Cert);
 	RsPgpId pgpId(sslCn);
 
+    RsPeerDetails det;
+    if(!rsPeers->getPeerDetails(sslId,det))
+    {
+        std::cerr << "Nothing known about peer " << sslId << " trying to connect! Refusing connection." << std::endl;
+        return verificationFailed;
+    }
+
+	bool isSslOnlyFriend = det.skip_pgp_signature_validation;
+
+    if(det.gpg_id != pgpId)
```

Comment:
This extra check deny connection to a new location of a PGP friend, IMHO we should remove it. PGP-id matching is correctly done at line 1275.

Reply:
not at all, because it should have the same PGP id.

Reply:
the problem is that `det` is not populated if sslid is not known